symengine/symengine_rcp.h

```diff
@@ -198,6 +198,73 @@ using Teuchos::print_stack_on_segfault;
 #endif
 
 
+template <class T>
+class EnableRCPFromThis {
+// Public interface
+public:
+    //! Get RCP<T> pointer to self (it will cast the pointer to T)
+    inline RCP<T> rcp_from_this() {
+#if defined(WITH_SYMENGINE_RCP)
+        return rcp(static_cast<T*>(this));
+#else
+        return rcp_static_cast<T>(weak_self_ptr_.create_strong());
+#endif
+    }
+
+    //! Get RCP<const T> pointer to self (it will cast the pointer to const T)
+    inline RCP<const T> rcp_from_this() const {
```

Comment:
I think that the compiler will pick up the right version of these routines, based on const. I could be wrong though.